lib/aggregation/dataflow-reducer-config-factory/src/test/aggregator-config-test.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+const { extend, omit } = require('underscore');
+const { createAggregatorConfig } = require('../lib/aggregator-config');
+
+describe('aggregator reducer config test', () => {
+  const dedupId = 'dedup-id';
+
+  let udoc = {
+    organization_id: 'org-id',
+    space_id: 'space-id',
+    resource_instance_id: 'res-instance-id',
+    consumer_id: 'consumer-id',
+    plan_id: 'plan-id',
+    metering_plan_id: 'metering-plan-id',
+    pricing_plan_id: 'pricing-plan-id',
+    rating_plan_id: 'rating-plan-id'
+  };
+
+  let reducerConfig;
+  let sandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  const validateReducerConfig = (msg, isSecured) => context(msg, () => {
+    beforeEach(() => {
+      reducerConfig = createAggregatorConfig(sandbox.stub().returns(isSecured), sandbox.any, sandbox.any);
+    });
+
+    it('correct write scopes are returned', () => {
+      expect(reducerConfig.input.wscope(udoc)).to.deep.equal(isSecured ? {
+        system: ['abacus.usage.write']
+      } : undefined);
+    });
+
+    it('correct read scopes are returned', () => {
+      expect(reducerConfig.input.rscope(udoc)).to.deep.equal(isSecured ? {
+        system: ['abacus.usage.read']
+      } : undefined);
+    });
+
+    context('consumer id is missing', () => {
+      const udocWithoutConsumer = omit(udoc, 'consumer_id');
+
+      it('correct input key is returned', () => {
+        expect(reducerConfig.input.key(udocWithoutConsumer))
+          .to.equal('org-id');
+      });
+
+      it('correct input groups is returned', () => {
+        expect(reducerConfig.input.groups(udocWithoutConsumer)[0]).to.equal(
```

Comment:
Consider comparing arrays directly:
expect(reducerConfig.input.groups(udocWithoutConsumer)).to.equal(['org-id', '....', '...'])

Reply:
Done using `to.deep.equal`